microsoft/testsuites/dpdk/dpdkutil.py

```diff
@@ -546,7 +475,7 @@ def verify_dpdk_send_receive(
     test_duration: int = variables.get("dpdk_test_duration", 15)
     kill_timeout = test_duration + 5
     test_kits = init_nodes_concurrent(
```

Comment:
remember that if you're going to raise exceptions if the hugepage allocation fails; you'll need to catch and handle here. We should catch NotEnoughMemoryException and raise a SkippedException(err) that wraps the that exception. Any others; we should allow to bubble up since they imply bugs.

Reply:
Kindly clarify on why we shall skip if there is not memory? If there is not enough memory there has to be a test bug at least to perform an environment change or any other decision to be taken up.
I intentionally kept it like that so that the test fails and brings it to our notice.

Reply:
If there isn't enough memory, we're running on a system which doesn't support that test. We cannot determine if DPDK is working because the environment doesn't have enough resources to even start DPDK. There are a lot of sizes of VMs and machines which won't have enough memory for the test; that's alright. We just don't want to run the test on those. We **will** notice if all the DPDK tests get skipped with a NotEnoughMemory error. We don't want that polluting test results, because the test **did not fail**; rather, the test did not run enough to start DPDK.

Reply:
Same behavior with trying to run a given hugepage size test on a given platform. If the architecture doesn't support a certain hugepage size; we shouldn't fail the test. It's a skip. It's not a 'PASS' because we didn't test anything. It's not a FAIL either.

Reply:
Are we sure that we won't come across a scenario where the setup actually supports yet doesn't have enough memory to carry out the operation?
Clarifying with you because this is a separate tool that we are writing here, and there can be tests with many scenarios.
Do let me know your thought on this question

Reply:
discussed offline, made the change to skip it